### ISSUE TITLE
How do folks handle labs?

### ISSUE THREAD

**Issue by karafecho** [NCATSTranslator/Clinical-Data-Committee-Tracking-Voting#3]
This issue is intended to initiate a discussion regarding the treatment of clinical laboratory measurements within the context of Translator.

**Comment by karafecho** [comment]
The ICEES+ Exposures Provider considers LOINC codes + flags (standardized to "high", "low", "normal") to provide meaning/interpretation. LOINC codes are rolled up such that any LOINC related to, say, ALT is treated the same, with the flags providing the interpretation. The service itself exposes LOINC IDs, SNOMED IDs, UMLS IDs, etc. Presumably, the flags will at some point be exposed as metadata via a TRAPI response. We have not yet mapped to HPO, but are considering this. Interested in hearing how others are treating labs.

**Comment by cbizon** [comment]
It seems like annotating these labs with HPO (and/or other non-LOINC things) would be very helpful with integrating into Translator.

But, as I guess most folks know, HPO and LOINC id's are not equivalent, even where there is a mapping.  It's something more like LOINC + as specific result = HPO.  

It seems like that kind of mapping should be done once somewhere, maybe exposed as a KP or part of a KP?

**Comment by karafecho** [comment]
Yes, and we (Exposures Provider) are conceptualizing the implementation as LOINC + flag = HPO.

**Comment by CaseyTa** [comment]
Lab tests and their results are currently not available in COHD, but we have a similar plan to what Kara already mentioned. We've made some efforts into applying LOINC2HPO (with help from OMOP2OBO for applying the LOINC2HPO mappings to our OMOP database), but we've had limited success so far in terms of 1) the number of phenotypes we're finding per patient (much lower than what the UNC folks saw in their publication) and 2) accuracy of the resulting HPO concepts.

**Comment by karafecho** [comment]
Good to know, Casey.

The initial LOINC2HPO mappings were developed using lab data from UNC's asthma cohort, so I'm not surprised by (1), but (2) is concerning.

**Comment by karafecho** [comment]
Summary from 03.12.21 meeting, with a few additional thoughts:

1. We all agree that mapping LOINCs + flags to HPO (or possibly UMLS) is a useful activity, one that will benefit the Translator program and increase the appeal of the clinical KPs.
2. We further agree that a simply lossy approach with less precise mappings and a finite set of labs/phenotypes makes sense, at least initially.
3. Open questions include:

- Should the clinical KPs or some subset or (perhaps) the SRI create a new Lab Provider that returns HPO IDs given a LOINC + flag? Or, should the clinical KPs perform the mappings independently?
- Should we tackle this issue prior to the December 2021 demo, or defer until the next funding segment?

**Comment by jh111** [comment]
Also see meeting notes for additional details.

General note, consider tackling this complex problem in steps over time, while addressing other high-yield priorities.

Specific issues include the following.
1. Many-to-many maps for labs to phenotype
2. The fact a lab was taken (which may indicate ruling out a different disease than the patient has).
3. Whether a lab is above or below: only some labs have reference ranges, and those may vary significantly by age, gender, and personal baseline). LOINC2HPO

Ways to address issues.
- Proceed with LOINC2HPO, and/or other non-curated solutions, and add clear provenance (warnings).
- Consider adding maps to chemicals, proteins (need to find source for this).
- Where possibly directly, maps from labs to drugs (ie: vancomycin trough)
- Add curated edges for a limited set of common labs. See examples. Ignore the ranges (use range from the EHR lab, or the flg), but consider these to be more likely to have universal relevance. ABIM. [https://www.abim.org/~/media/ABIM%20Public/Files/pdf/exam/laboratory-reference-ranges.pdf](url)

**Comment by karafecho** [comment]
A few quick thoughts, @jh111 :

1. Re "The fact a lab was taken (which may indicate ruling out a different disease than the patient has)", I'm not sure this matters, at least not for Translator purposes. The fact that a lab was taken indicates suspicion, so isn't that valuable information?
2. Re "Whether a lab is above or below: only some labs have reference ranges, and those may vary significantly by age, gender, and personal baseline)", while this is absolutely true, if we rely on flags (at least initially), then reference ranges don't really matter because flags provide an interpretation of measurement values wrt reference ranges for any given patient. Flags seem like the path of least resistance in order to move this effort along.
3. Re "add clear provenance (warnings)", yes, most definitely.
4. Re Consider adding maps to chemicals, proteins", I may be missing the point here, but I don't see any value in this effort. Wouldn't such a mapping just lead back to "was something tested or not"?
5. Re "maps from labs to drugs", are you suggesting potential drug treatments for abnormal labs?
6. Re "Add curated edges for a limited set of common labs", yes, let's start with a limited set of common labs.

Thanks!
